telemetry/data_window.go

```diff
@@ -80,6 +110,124 @@ var (
 	subWindowsLock    = sync.RWMutex{}
 )
 
+func getSQLSum(sqlTypeDeta *sqlType) int64 {
+	result := sqlTypeDeta.Use
+	result += sqlTypeDeta.Show
+	result += sqlTypeDeta.Begin
+	result += sqlTypeDeta.Commit
+	result += sqlTypeDeta.Rollback
+	result += sqlTypeDeta.Insert
+	result += sqlTypeDeta.Replace
+	result += sqlTypeDeta.Delete
+	result += sqlTypeDeta.Update
+	result += sqlTypeDeta.Select
+	result += sqlTypeDeta.Other
+	return result
+}
+
+func readSQLMetric(timepoint time.Time, SQLresult *sqlUsageData) error {
+	ctx := context.TODO()
+	result, err := querySQLMetric(ctx, timepoint)
+	if err != nil {
+		if err1, ok := err.(*promv1.Error); ok {
+			return errors.Errorf("query metric error, msg: %v, detail: %v", err1.Msg, err1.Detail)
+		}
+		return errors.Errorf("query metric error: %v", err.Error())
+	}
+
+	dataAnylis(result, SQLresult)
+	return nil
+
+}
+
+func querySQLMetric(ctx context.Context, queryTime time.Time) (result pmodel.Value, err error) {
+	// Add retry to avoid network error.
+	var prometheusAddr string
+	for i := 0; i < 5; i++ {
+		//TODO: the prometheus will be Integrated into the PD, then we need to query the prometheus in PD directly, which need change the quire API
+		prometheusAddr, err = infosync.GetPrometheusAddr()
+		if err == nil || err == infosync.ErrPrometheusAddrIsNotSet {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	if err != nil {
+		return nil, err
+	}
+	promClient, err := api.NewClient(api.Config{
+		Address: prometheusAddr,
+	})
+	if err != nil {
+		return nil, err
+	}
+	promQLAPI := promv1.NewAPI(promClient)
+	ctx, cancel := context.WithTimeout(ctx, promReadTimeout)
+	defer cancel()
+	promQL := "sum(tidb_executor_statement_total{}) by (instance,type)"
+	// Add retry to avoid network error.
+	for i := 0; i < 5; i++ {
+		result, _, err = promQLAPI.Query(ctx, promQL, queryTime)
+		if err == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return result, err
+}
+
+func dataAnylis(promResult pmodel.Value, SQLresult *sqlUsageData) {
```

Comment:
dataAnylis -> anylisSQLUsage

Reply:
Thank you for your commet, I will accept it!